lightning-background-processor/src/lib.rs

```diff
@@ -360,12 +376,24 @@ macro_rules! define_run_body {
 				break;
 			}
 
+			if $timer_elapsed(&mut last_forwards_processing_call, cur_batch_delay) {
+				$channel_manager.get_cm().process_pending_htlc_forwards();
```

Comment:
Looked a bit closer at this function. There is a lot of logic in there. Also various locks obtained.